### PR TITLE
Adds logic for BYE Scenarios

### LINKED ISSUE #492: Add conditional message for BYE with remaining balance scenario
### Description

UIO will have a new boolean flag via the API for claimants that are past the benefit year (BYE) and have a remaining balance. UIB will share the appropriate content and conditional logic for us to implement.

### Acceptance Criteria

- [ ] Pair on logic flow updates
- [ ] Design to mock content structure if needed
- [ ] Eng to implement logic updates
- [ ] Pair on testing as needed

### PR DESCRIPTION
**Pull request by lomky**
## Changes
Adds the BYE scenarios as defined [here on 2021-10-13](https://docs.google.com/spreadsheets/d/1tQEUjk4o_cvW5kStd-f1i0l-JN2cGzXV/edit#gid=1265468978) with the business logic flow [as defined & agreed on](https://lucid.app/lucidchart/e0d100d3-44aa-48c6-8888-f48079c93019/edit?page=SqiUDMrSxKq6&invitationId=inv_d4635b74-95b8-4834-81d8-fe6e1446e692#)

BYE is :
- lower priority than all other scenarios (except `Base State without Weeks to Certify`/`scenario 5`)
- triggers for the Program Types:
  - UI: `Scenario 7`
  - EUC, PEUC, EUX, PEUX, EUY, PEUY, EUW, EUZ, FEDED: `Scenario 8`
  - PUA: `Scenario 9`
  - DUA: `Scenario 10`

Content in this PR is draft only

## Testing
  - Lacking any UniqueNumbers for the scenarios, test via the Page Storybook's scenario Dropdown

## Follow up tickets
 - [x] file issue to remove deprecated hasPendingWeeks field post-R2
 - [x] file issue to refactor redundant & fragile lists of Scenarios
 - [x] file issue for final content
 
===

Resolves #492 

- [x] Implement BYE logic

### CODE REVIEW

File: tests/utils/getScenarioContent.test.ts [hunk 1]
```diff
@@ -66,6 +66,21 @@ describe('The Base State scenarios (scenarios 5 & 6)', () => {
...
+      ScenarioType.Scenario7,
+      ScenarioType.Scenario8,
+      ScenarioType.Scenario9,
+      ScenarioType.Scenario10,
+    ]
+    for (const scenarioType of byeScenarios) {
+      const scenarioObject = getScenario(apiGatewayStub(scenarioType))
+      expect(scenarioObject.scenarioType).toBe(scenarioType)
+    }
+  })
```
Comment (lomky) [inline tests/utils/getScenarioContent.test.ts:81]: Use `it.each`

File: types/common.ts [hunk 2]
```diff
@@ -49,8 +49,10 @@ export interface Claim {
   ClaimType?: null | undefined | string
   uniqueNumber?: null | string
   claimDetails?: null | ClaimDetailsResult
-  hasPendingWeeks?: null | undefined | boolean
+  hasPendingWeeks?: null | undefined | boolean // deprecated for hasValidPendingWeeks
+  hasValidPendingWeeks?: null | undefined | boolean
   hasCertificationWeeksAvailable?: null | undefined | boolean
+  isBye?: null | undefined | boolean
```
[the remark this answers is not in the document]
Comment (lomky) [reply types/common.ts:55]: I think we won't know until they give us the API response for us to test against. How about I file this as a follow on, triggered to start when we get the API response?

[the remark this answers is not in the document]
Comment (rocketnova) [reply types/common.ts:55]: Sounds good to me!

File: tests/utils/getScenarioContent.test.ts [hunk 3]
```diff
@@ -57,12 +57,26 @@ describe('The Generic Pending scenario (scenario 4)', () => {
...
+describe('The BYE scenarios (scenarios 7, 8, 9, 10)', () => {
+  const byeScenarios = [
+    ['UI', ScenarioType.Scenario7],
+    ['Federal Extension', ScenarioType.Scenario8],
+    ['PUA', ScenarioType.Scenario9],
+    ['DUA', ScenarioType.Scenario10],
+  ]
+  it.each(byeScenarios)('BYE for %s returns as expected', (description, scenarioType) => {
+    const scenarioObject = getScenario(apiGatewayStub(scenarioType))
+    expect(scenarioObject.scenarioType).toBe(scenarioType)
```
[the remark this answers is not in the document]
Comment (lomky) [reply tests/utils/getScenarioContent.test.ts:79]: I don't think thats overkill at all! ty for catching

File: tests/utils/getScenarioContent.test.ts [hunk 4]
```diff
@@ -57,12 +57,83 @@ describe('The Generic Pending scenario (scenario 4)', () => {
...
+    ['state extension', 'CAL-ED Extension'],
+    ['training extension', 'Training Extension (TE)'],
+  ]
+
+  it.each(nonByeProgramTypes)('BYE is not triggered for a %s', (description, programType) => {
+    // we're overrding the program type, which is the difference between the API response
+    // for scenarios 7/8/9/10 - so testing each would be redundant
+    const byeInvalidProgram = apiGatewayStub(ScenarioType.Scenario7)
+    byeInvalidProgram.claimDetails.programType = programType
+    byeInvalidProgram.claimDetails.programType = programType
```
Comment (rocketnova) [inline tests/utils/getScenarioContent.test.ts:133]: I think you have duplicate lines here.

**Review by rocketnova — APPROVED** [review]
LGTM!

**Pull request merged**